From a9613ae85cb6288e680599943d9e47a1f3d4f75a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 19 Aug 2018 19:40:21 +0300
Subject: [PATCH] Makefile: add gaudi/dali-release target

The new target somewhat automates the release preparation process.
The process is defined as:
1) Take the numeric version from command line (env variable)
2) Create new tag (depending on the target) with the provided version
3) Push the tag upstream (origin)

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index 127ace8..19ce7f5 100644
--- a/Makefile
+++ b/Makefile
@@ -24,6 +24,13 @@
 	@echo '                           set DALI_INSTALL_HOST_IP=<host IP> to'
 	@echo '                             override <board> default host IP'
 	@echo
+	@echo 'Release:'
+	@echo '  required environment variables:'
+	@echo '    HL_RELEASE_VER     - Habanalabs release version string'
+	@echo '  dali-release         - Prepare the release tag and push it upstream'
+	@echo '                           usually, the version string would be something'
+	@echo '                             like: x.y.z'
+	@echo
 	@echo 'Testing:'
 	@echo '  dali-<board>-test    - Run uboot test on <board> e.g. onix, pldm, asic'
 	@echo '  dali-<board>-tbot    - Run tbot test on <board> e.g. onix, pldm, asic'
@@ -48,6 +55,22 @@
 include $(HL_VARS_FILE)
 endif
 
+.PHONY: hl-check-release
+hl-check-release:
+ifndef HL_RELEASE_VER
+	$(error HL_RELEASE_VER is undefined)
+endif
+
+RELEASE_REPOS := u-boot linux
+
+.PHONY: gaudi-release dali-release
+gaudi-release dali-release: hl-check-release
+	$(eval HL_FULL_VER=hl-$(patsubst %-release,%,$@)-$(HL_RELEASE_VER))
+	@repo forall $(RELEASE_REPOS) -c \
+		git tag -afm "Release Habanalabs $(HL_FULL_VER)" $(HL_FULL_VER)
+	@repo forall $(RELEASE_REPOS) -c git push origin $(HL_FULL_VER)
+	@echo "Released: $(HL_FULL_VER)"
+
 .PHONY: hl-dali hl-gaudi
 hl-dali hl-gaudi: $(OUTPUT_DIR)
 	@echo "export HL_DEV=$(patsubst hl-%,%,$@)"		 > $(HL_VARS_FILE)
